04/dbernhard/advanced_while.oat

```diff
@@ -0,0 +1,9 @@
+int program(int argc, string[] args) {
+    var x = 10;
+
+    while (x) {
```

Comment:
*x* is of type *int* and therefore cannot be used as a condition of while.

I think you can change it to the explicit *x >= 0* or *x != 0*

Reply:
You are right, i overlooked the TYP_WHILE rule.